pkg/builder/builder.go

```diff
@@ -866,7 +866,7 @@ func BuildVProxyDeployment(nm types.NamespacedName, vdb *vapi.VerticaDB, sc *vap
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        nm.Name,
 			Namespace:   nm.Namespace,
-			Labels:      MakeLabelsForVProxyObject(vdb, sc, true),
+			Labels:      MakeLabelsForVProxyObject(vdb, sc, false),
```

Comment:
Change `forPod ` to false to add label `vertica.com/subcluster-name` to the proxy deployment. Only objects with `vertica.com/subcluster-name` will be deleted In `checkForDeletedSubcluster`.

Labels added to the object when forPod is true:
```
    vertica.com/subcluster-selector-name: v-client-proxy-sec1
    vertica.com/subcluster-svc: sec1
```
Labels added to the object when forPod is false:
```
    vertica.com/subcluster-name: sec1
    vertica.com/subcluster-type: secondary
```